x-pack/packages/ml/error_utils/src/types.ts

```diff
@@ -35,6 +35,7 @@ export interface MLResponseError {
 }
 
 export interface ErrorMessage {
+  query: string;
```

Comment:
It looks like we have two different versions of `ErrorMessage` in ML, this one is used in conjunction with `isErrorMessage` and is used for extracting the message string from an error returned from kibana server which only contains a `message`.
The other types which contain `query` are created in [client side code,](https://github.com/elastic/kibana/blob/3c1c73f573d3f7b0b3890d703c491e9cfcdd951a/x-pack/plugins/ml/public/application/explorer/components/explorer_query_bar/explorer_query_bar.tsx#L163) where the `query` is added to the error object for later use.

I think we should keep these separate and have an interface which extends `ErrorMessage` to add the `query`.
IMO this new interface shouldn't live in this file as these types are purely for describing errors returned from es or kibana.

Reply:
Good point, I'll do some renaming to make them more distinct.